src/parsers/class-wxr-parser.php

```diff
@@ -11,23 +11,46 @@
  */
 class WXR_Parser {
 	public function parse( $file ) {
-		// Attempt to use proper XML parsers first
-		if ( extension_loaded( 'simplexml' ) ) {
-			$parser = new WXR_Parser_SimpleXML();
-			$result = $parser->parse( $file );
-
-			// If SimpleXML succeeds or this is an invalid WXR file then return the results
-			if ( ! is_wp_error( $result ) || 'SimpleXML_parse_error' != $result->get_error_code() ) {
-				return $result;
+		// Allow forcing a specific parser via WXR_PARSER: simplexml|xml|regex|xmlprocessor
+		$preferred_parser = defined( 'PREFERRED_WXR_PARSER' ) ? constant( 'PREFERRED_WXR_PARSER' ) : null;
+		if ( $preferred_parser ) {
+			$available_parsers = array(
+				'simplexml'    => 'WXR_Parser_SimpleXML',
+				'xml'          => 'WXR_Parser_XML',
+				'regex'        => 'WXR_Parser_Regex',
+				'xmlprocessor' => 'WXR_Parser_XML_Processor',
+			);
+			if ( isset( $available_parsers[ $preferred_parser ] ) ) {
+				$parser = new $available_parsers[ $preferred_parser ]();
+				$result = $parser->parse( $file );
+			} else {
+				_doing_it_wrong( __FUNCTION__, sprintf( __( 'Invalid parser specified: %s', 'wordpress-importer' ), $preferred_parser ), '0.7.0' );
```

Comment:
This should be `0.8.3`. This PR has functional changes, so a new version should be released to dotorg.

I suggest setting `0.9.0` here. We can land this and https://github.com/WordPress/wordpress-importer/pull/199 and bump the version in the other PR.